util/codec/bytes.go

```diff
@@ -122,6 +122,27 @@ func DecodeBytesDesc(b []byte) ([]byte, []byte, error) {
 	return decodeBytes(b, true)
 }
 
+// EncodeCompactBytes joins bytes with its length into a byte slice. It is more
+// efficient in both space and time compare to EncodeBytes.
+func EncodeCompactBytes(b []byte, data []byte) []byte {
+	b = EncodeVarint(b, int64(len(data)))
+	return append(b, data...)
+}
+
+// DecodeCompactBytes decodes bytes which is encoded by EncodeCompactBytes before.
+func DecodeCompactBytes(b []byte) ([]byte, []byte, error) {
+	b, n, err := DecodeVarint(b)
+	if err != nil {
+		return nil, nil, errors.Trace(err)
+	}
+	if int64(len(b)) < n {
+		return nil, nil, errors.Errorf("insufficient bytes to decode value, expected length: %v", n)
+	}
+	data := make([]byte, int(n))
```

Comment:
The safety depends on how to use it. As long as the caller discards original `b` after calling, returning `b[:n]` directly is safe. So it is safe for now, but we should take care of the usage in the future.